core/frontend/src/IModelConnection.ts

```diff
@@ -907,6 +922,62 @@ export namespace IModelConnection { // eslint-disable-line no-redeclare
       const iModel = this._iModel;
       return iModel.isOpen ? IModelReadRpcInterface.getClientForRouting(iModel.routingContext.token).queryElementProps(iModel.getRpcProps(), params) : [];
     }
+
+    public async getPlacements(elementIds: Iterable<Id64String>): Promise<Array<Placement & { elementId: Id64String }>> {
+      let ids: Id64String[];
+      if (typeof elementIds === "string")
+        ids = [elementIds];
+      else if (!Array.isArray(elementIds))
+        ids = Array.from(elementIds);
+      else
+        ids = elementIds;
+
+      if (ids.length === 0)
+        return [];
+
+      const ecsql = `
+        SELECT
+          ECInstanceId,
+          Origin.x as x, Origin.y as y, Origin.z as z,
+          BBoxLow.x as lx, BBoxLow.y as ly, BBoxLow.z as lz,
+          BBoxHigh.x as hx, BBoxHigh.y as hy, BBoxHigh.z as hz,
+          Yaw, Pitch, Roll,
+          NULL as Rotation
+        FROM bis.GeometricElement3d WHERE Origin IS NOT NULL AND BBoxLow IS NOT NULL AND BBoxHigh IS NOT NULL
+        UNION ALL
+        SELECT * FROM (
+          SELECT
+            ECInstanceId,
+            Origin.x, Origin.y, NULL,
+            BBoxLow.x, BBoxLow.y, NULL,
+            BBoxHigh.x, BBoxHigh.y, NULL,
+            NULL, NULL, NULL,
+            Rotation
+          FROM bis.GeometricElement2d WHERE Origin IS NOT NULL AND BBoxLow IS NOT NULL AND BBoxHigh IS NOT NULL)
+        WHERE ECInstanceId IN (${ids.join(",")})
+      `;
+
+      const placements = new Array<Placement & { elementId: Id64String}>();
+      for await (let row of this._iModel.query(ecsql)) {

```

Comment:
should this use the quota and limit stuff? I don't know how that works.

Reply:
As far as I'm aware @khanaffan imposes default quotas/limits. Certainly can't expect everybody else to come up with sensible values.

Reply:
looks to me like he doesn't limit by default, which seems wrong to me.

Reply:
I don't think he should add a LIMIT by default.
The `query` method is an async iterator that is supposed to page results, as opposed to `queryRows` which returns all the results in one response.